test/Mono.Linker.Tests.Cases/Libraries/RootLibrary.cs

```diff
@@ -214,11 +214,13 @@ public class UninstantiatedPublicClassWithInterface :
 			[Kept]
 			public void InternalInterfaceMethod () { }
 
+			[Kept]
```

Comment:
We're probably missing a test here (from before).
Before this change something like:
```C#
internal interface IFace
{
    void Method();
}

internal class TestType : IFace
{
    IFace.Method() { }
}

void Test()
{
    new TestType();
    // IFace is never mentioned anywhere
}
```

In this case we should have removed IFace as a whole. Now we should keep it.
I was expecting to see a test which would fail on this (or rather that you had to change it, but I don't see that anywhere).

Reply:
If I understand correctly, `IInternalInterface` was essentially the same situation, but we never got rid of it because with the `UnusedInterfaces` optimization off in library mode we had to keep the interface and the interface implementation.

Reply:
Ah - so it's the combination of two things which basically makes this untestable... OK I guess it makes sense.